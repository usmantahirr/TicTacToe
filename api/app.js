const express = require("express");
const path = require("path");
const cookieParser = require("cookie-parser");
const logger = require("morgan");
const cors = require('cors');
const gamesRouter = require("./routes/games");

const app = express();

app.use(cors());
app.use(logger("dev"));
app.use(express.json());
app.use(express.urlencoded({ extended: false }));
app.use(cookieParser());
app.use(express.static(path.join(__dirname, "public")));

app.use("/api/games", gamesRouter);

module.exports = app;
