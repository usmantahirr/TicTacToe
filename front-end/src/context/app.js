import React, { useContext } from "react";

const AppContext = React.createContext({
  activeGame: {},
  setActiveGame: () => {}
});

export const useAppContext = () => useContext(AppContext);
export default AppContext;
