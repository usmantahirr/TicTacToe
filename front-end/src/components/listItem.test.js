import React from "react";
import * as AppContext from "../context/app";
import { shallow } from "enzyme";
import ListItem from "./listItem";

describe("List Item", () => {
  let component;
  const setActiveGame = jest.fn();

  beforeAll(() => {
    const context = {
      activeGame: {
        id: 1,
        name: "example"
      },
      setActiveGame
    };

    jest.spyOn(AppContext, "useAppContext").mockImplementation(() => context);

    component = shallow(
      <ListItem
        game={{
          id: 1,
          name: "example",
          createdAt: new Date('05/05/2020').valueOf(),
          updatedAt: new Date('05/06/2020').valueOf()
        }}
      />
    );
  });

  it("Should call set active function from context", () => {
    component.find("IconButton").simulate("click");
    expect(setActiveGame).toHaveBeenCalled();
  });

  it("should pass required values in context", () => {
    expect(component.find("span").text()).toBe("example");
  });

  it("Should render correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
