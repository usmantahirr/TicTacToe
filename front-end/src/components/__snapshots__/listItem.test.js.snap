// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item Should render correctly 1`] = `
<styled.div
  className="active"
>
  <span>
    example
  </span>
  <styled.span>
    5/5/2020
  </styled.span>
  <styled.span>
    12:00:00 AM
  </styled.span>
  <div
    style={
      Object {
        "float": "right",
      }
    }
  >
    <IconButton
      onClick={[Function]}
      style={
        Object {
          "marginRight": "5px",
        }
      }
    >
      <ArrowRight
        color="currentColor"
        size={15}
      />
    </IconButton>
  </div>
</styled.div>
`;
